editor/utils/url.js

```diff
@@ -18,3 +19,32 @@ export function addQueryArgs( url, args ) {
 
 	return format( { ...parsedUrl, query } );
 }
+
+/**
+ * Returns the Gutenberg page URL with extra query strings
+ *
+ * @param  {Object} query  Query Args
+ *
+ * @return {String}        URL
+ */
+export function getGutenbergURL( query = {} ) {
+	const [ baseUrl, currentQuery ] = window.location.href.split( '?' );
+	const qs = parseQueryString( currentQuery || '' );
```

Comment:
Minor: The default can be assigned in the array destructure for clarity/consistency:

```js
const [ baseUrl, currentQuery = '' ] = window.location.href.split( '?' );
```